### PR TITLE
Added string.d and updated README

### PR DESCRIPTION
**Pull request by JackStouffer**
Added the planned to be deprecated functions from std.string

### CODE REVIEW

**Comment by JackStouffer** [comment]
https://github.com/dlang/phobos/pull/5378

**Comment by wilzbach** [comment]
FYI: Travis is failing with:
```
undead 1.0.6+commit.15.g7f04928: building configuration "undead-test-library"...
../../../../../tmp/dub_test_root-d5433945-da6d-4616-91eb-3a4e30701d31.d(11,15): Error: module pattern from file src/undead/pattern.d must be imported with 'import pattern;'
dmd failed with exit code 1.
```

**Comment by JackStouffer** [comment]
@wilzbach Fixed

**Comment by CyberShadow** [comment]
Not sure about the file name. I think the structure should mirror that of Phobos', plus `s/^std\./undead./`. Meaning, this should go into a new `string.d` file, which could then receive further additions should we decide to deprecate more things from std.string.

**Comment by JackStouffer** [comment]
@CyberShadow Changed

File: README.md [hunk 1]
```diff
@@ -13,3 +13,9 @@ Current modules included:
 * std.dateparse
 * std.regexp
 * std.stream and friends
+
+Some individual functions have been moved here prior to their removal from
+Phobos. They are
+
+* undead.doformat: Contains the `doFormat` function from std.format
```
Comment (wilzbach) [inline README.md]: `doFormat` has been removed from Phobos?
https://github.com/dlang/phobos/commit/cf65a3be036ecfb92c0b5b6f17d77a2a2c58e82a

Comment (JackStouffer) [inline README.md]: Yeah the wording here is a bit confusing. Will change

**Comment by JackStouffer** [comment]
@CyberShadow Other PR is now merged, let's pull this

**Pull request merged**